lib/core/heartbeat.rb

```diff
@@ -6,62 +6,24 @@
 
 # frozen_string_literal: true
 
-require 'concurrent'
 require 'connectors/connector_status'
 require 'connectors/registry'
 require 'core/connector_settings'
 require 'core/elastic_connector_actions'
-require 'utility/logger'
 
 module Core
   class Heartbeat
-    INTERVAL_SECONDS = 60
-
     class << self
-      def start_task(connector_id, service_type)
-        Utility::Logger.debug("Starting heartbeat timer task for [#{connector_id} - #{service_type}] with interval #{INTERVAL_SECONDS} seconds.")
-
-        Concurrent::TimerTask.execute(execution_interval: INTERVAL_SECONDS, run_now: true) do
-          Utility::Logger.debug("Sending heartbeat for the connector [#{connector_id} - #{service_type}].")
-          send(connector_id, service_type)
-        rescue StandardError => e
-          Utility::ExceptionTracking.log_exception(e, 'Heartbeat timer encountered unexpected error.')
-        end
-
-        Utility::Logger.info("Successfully started heartbeat task for the connector [#{connector_id} - #{service_type}].")
-      end
-
-      def send(connector_id, service_type)
-        connector_settings = Core::ConnectorSettings.fetch_by_id(connector_id)
-
+      def send(connector_settings)
         doc = {
-          :last_seen => Time.now
+            :last_seen => Time.now
         }
-
-        if connector_settings.connector_status == Connectors::ConnectorStatus::CREATED
-          connector_class = Connectors::REGISTRY.connector_class(service_type)
-          configuration = connector_class.configurable_fields
-          doc[:service_type] = service_type
-          doc[:configuration] = configuration
-
-          # We want to set connector to CONFIGURED status if all configurable fields have default values
-          new_connector_status = if configuration.values.all? { |setting| setting[:value].present? }
-                                   Utility::Logger.debug("All connector configurable fields provided default values for connector #{connector_id}.")
-                                   Connectors::ConnectorStatus::CONFIGURED
-                                 else
-                                   Connectors::ConnectorStatus::NEEDS_CONFIGURATION
-                                 end
-
-          doc[:status] = new_connector_status
-
-          Utility::Logger.info("Heartbeat updated configuration for connector #{connector_id}.")
-          Utility::Logger.info("Changing connector status to #{new_connector_status}.")
-        elsif connector_settings.connector_status_allows_sync?
-          connector_instance = Connectors::REGISTRY.connector(service_type, connector_settings.configuration)
+        if connector_settings.connector_status_allows_sync?
```

Comment:
I think this part can go to Configuration task too, leaving heartbeat to only do this small `last_seen` update.

I think this update of status of connector with heartbeat is a bit confusing and led me to really awkward UI interactions. (Heartbeat thread decided that connector is not connected due to small network lag, but connector was running in the meantime, so connector was marked as ERROR, while it was working actually.)

Reply:
This basically checks the connection to remote source, so I think it falls well in the scope of heartbeat?

>Heartbeat thread decided that connector is not connected due to small network lag

I would see chances are really small as we have retry enabled in our ES client.

Reply:
> This basically checks the connection to remote source, so I think it falls well in the scope of heartbeat?

I think it depends on our view here, idk how much customer value it brings in terms of UX. 

I just had a feeling that making heartbeat write-only (no read operation) might be nice, but I think I might be overthinking this.

Reply:
I don't even think it's necessary to report remote source status in heartbeat. We only care about remote source status when we want to sync data. This is how Workplace Search works.